data/Templates/Hl7v2/ADT_A01.liquid

```diff
@@ -2,14 +2,34 @@
     "resourceType": "Bundle",
     "type": "transaction",
     "entry": [
-        {% assign firstSegments = hl7v2Data | get_first_segments: 'PID|PD1|PV1|PV2|PR1|AVR' -%}
+        {% assign firstSegments = hl7v2Data | get_first_segments: 'PID|PD1|PV1|PV2|PR1|AVR|MSH' -%}
+
+        {% evaluate messageHeaderId using 'ID/MessageHeader' MSH: firstSegments.MSH -%}
+        {% if messageHeaderId -%}
+            {% include 'Resource/MessageHeader' MSH: firstSegments.MSH, ID: messageHeaderID -%}
+        {% endif -%}
+
+        {% evaluate provenanceId using 'ID/Provenance' MSH: firstSegments.MSH, baseID: patientId -%}
```

Comment:
We also need to update the variable in `ID/Provenance` from `ORC.Value` to `MSH.Value`. If we keep that template unchanged, `ID/Provenance` still expects ORC and will not be generated with MSH.

Reply:
Sure, I've updated the ID generation template.